core/trace/src/main/scala/org/typelevel/otel4s/trace/SpanBuilder.scala

```diff
@@ -253,7 +253,7 @@ object SpanBuilder {
       builder: SpanBuilder[F]
   )(implicit kt: KindTransformer[F, G])
       extends SpanBuilder[G] {
-    def meta: InstrumentMeta[G] =
+    def meta: InstrumentMeta.Dynamic[G] =
       builder.meta.mapK[G]
```

Comment:
hmmm... should this be a `val` so we're not computing/allocating the mapped instance repeatedly? (this applies to probably several places). also... isn't technically related to this PR, and *I'm* the one who did it that way originally lol

Reply:
Yeah, that's valid. In the following case, we will 'mapK' the underlying meta at least 2 times:
```scala
val tf = Tracer[F]
val tg = Tracer[F].mapK[G]
tg.spanBuilder("builder").addAttributes(...).withKind(...).build 
```